### ISSUE TITLE
Automated detection of unwanted images (including unwanted objects e.g. tweezers)

### ISSUE THREAD

**Issue by PhilippMRehsen** [Aarhus-University-MPE/BioDiscover#20]
### Description of the new feature / enhancement

The imaging process with BioDiscover includes dropping specimen into the cuvette, which is for now done mostly manually. Specimen can occasionally get stuck on the site of the cuvette prior to being completely submerged in ethanol. With a manual mode of operation, this can get detected fast and is likely solved by pushing down the specimen into the cuvette with forceps or tweezers. During this scenario, it frequently happens that the tip of used tweezers are imaged (and measured). 

![1_#0000_24_2021_07_27-14-58-16-951](https://user-images.githubusercontent.com/106231532/181017520-811e7645-3b3f-47fd-b606-09e2e3454548.PNG)
Detecting intersections of the binary image with the edge of an image would also allow the detection of 

Suggested solution:
Automatically detect objects touching the upper edge of an image. This way "wrong" images wouldn't be taken or could be detected fast during post processing. E.g.: A column with something like "Object detected at edge" => "yes/no" could be used to filter images to either have a look at these images or exclude these images from further analyses. 

### Scenario when this would be used?

This features would be extremely helpful when dealing with small specimen, since they frequently get attached to the site of the cuvette.

### Supporting information

I could provide more images with forceps on them if needed.

**Comment by MadsR** [comment]
This is already part of the detection algorithm. The object must follow a set of criteria before cropping and saving begins:

- area > threshold
- Crop min > image min
- Crop max < Image max

The last two check for collisions with the edge, (see manual page 30).

Times where this check fail is when an object is separated into multiple blobs, as the checks are only done on the largest object.

**Multiple large objects in frame criteria**
A previous check was made that required a maximum of 1 "large" object in frame; if more than one blob met the area > threshold criteria, no cropping was made. This would previously fail, but could most likely be reimplemented to avoid "bad images".

**Comment by MadsR** [comment]
You can see the criteria described above in action, by starting image capture and lower a tweezer into the cuvette.

You should clearly see the tweezer in frame, and the object marked in red on the binary feed. But no images should be cropped and saved.

**Comment by PhilippMRehsen** [comment]
![1_Boye_052722_16_2_2022_05_30-14-19-20-548](https://user-images.githubusercontent.com/106231532/182843080-108c1e75-6fab-4315-b75b-8a7632a70862.PNG)
![1_Boye_052722_16_2_2022_05_30-14-19-20-587](https://user-images.githubusercontent.com/106231532/182843100-78afabe8-e592-4711-a590-c492f85c155b.PNG)
![1_Boye_052722_18_2_2022_05_30-14-24-47-973](https://user-images.githubusercontent.com/106231532/182843109-06fa24ce-3dff-4617-860d-158c4b218b7c.PNG)
![1_Boye_052722_18_2_2022_05_30-14-24-48-013](https://user-images.githubusercontent.com/106231532/182843119-68179abd-6e1f-4fa3-ae69-1be12ccfdb42.PNG)
![1_Boye_052722_18_2_2022_05_30-14-24-48-053](https://user-images.githubusercontent.com/106231532/182843129-d61a8171-3bca-43d2-8913-005f911c4d85.PNG)
![1_Boye_052722_18_2_2022_05_30-14-24-48-092](https://user-images.githubusercontent.com/106231532/182843145-d1b6310b-2ae4-4ea5-852c-d8c6f37d25b6.PNG)
![1_Boye_052722_19_2_2022_05_30-14-28-03-597](https://user-images.githubusercontent.com/106231532/182843151-801644cd-65e8-4ae1-87d8-90292f569317.PNG)
![1_Boye_052722_19_2_2022_05_30-14-28-03-635](https://user-images.githubusercontent.com/106231532/182843156-dce7c404-c94e-4a74-aa65-db5ff0fe65b8.PNG)
![Uploading 1_Boye_052722_19_2_2022_05_30-14-28-03-679.PNG…]()

**Comment by MadsR** [comment]
I agree the best solution to these would be to reimplement the "maximum large objects" criteria.

Image 1 and 3 does however appear to fail the edge detection; given how the tweezers appear as the largest object in frame and quite clearly touch the upper edge of the image.
